mapiproxy/servers/default/asyncemsmdb/dcesrv_asyncemsmdb.c

```diff
@@ -0,0 +1,1230 @@
+			obj_count = talloc_zero(data_pointers, uint32_t);
+			*obj_count = 0;
+			data_pointers[i] = obj_count;
+			retval = MAPI_E_SUCCESS;
+		}
+		else if (properties->aulPropTag[i] == PidTagLocalCommitTimeMax) {
+			/* TODO: temporary hack */
+			unix_time = time(NULL);
+			unix_to_nt_time(&nt_time, unix_time);
+			ft = talloc_zero(data_pointers, struct FILETIME);
+			MAPI_RETVAL_IF(!ft, MAPI_E_NOT_ENOUGH_MEMORY, NULL);
+			ft->dwLowDateTime = (nt_time & 0xffffffff);
+			ft->dwHighDateTime = nt_time >> 32;
+			data_pointers[i] = ft;
+			retval = MAPI_E_SUCCESS;
+		}
+		else {
+			retval = openchangedb_get_folder_property(data_pointers, openchangedb_ctx, p->username,
+								  properties->aulPropTag[i], folderId, data_pointers + i);
+		}
+		retvals[i] = retval;
+	}
+
+	return MAPI_E_SUCCESS;
+}
+
+
+
+/**
+   \details Process a TableModified event on a ContentsTable for row modified specific event type
+
+   \param mem_ctx pointer to the memory context
+   \param p pointer to the private asyncemsmdb data
+   \param s pointer to the array of subscriptions for this session
+   \param folderId the folder in which the tablemodified event occurred
+
+   \note This first iteration has the following limitations:
+   - Only TABLE_ROW_MODIFIED case are handled
+   - Only handle notifications within Inbox folder (openchangedb folders)
+
+   \return 0 on success, otherwise -1
+ */
+static int process_tablemodified_contentstable_notification(TALLOC_CTX *mem_ctx,
+							    struct asyncemsmdb_private_data *p,
+							    struct mapistore_notification_subscription *s,
+							    uint64_t folderId)
+{
+	int				i;
+	struct EcDoRpc_MAPI_REPL	reply;
+	DATA_BLOB			payload;
+	struct SPropTagArray		SPropTagArray;
+	void				**data_pointers;
+	enum MAPISTATUS			*retvals;
+	enum MAPISTATUS			retval;
+	int				ret;
+	uint32_t			prop_idx;
+	int				flagged = 0;
+	uint32_t			property;
+	void				*data = NULL;
+	enum ndr_err_code		ndr_err_code;
+	struct ndr_push			*ndr;
+	char				*mapistoreURI = NULL;
+
+	/* Looking */
+	for (i = 0; i < s->v.v1.count; i++) {
+		if (s->v.v1.subscription[i].flags & sub_TableModified) {
+			if ((s->v.v1.subscription[i].flags & sub_WholeStore) ||
+			    (s->v.v1.subscription[i].fid == folderId)) {
+				memset(&reply, 0, sizeof(struct EcDoRpc_MAPI_REPL));
+				reply.opnum = op_MAPI_Notify;
+				reply.error_code = MAPI_E_SUCCESS;
+				reply.u.mapi_Notify.NotificationHandle = s->v.v1.subscription[i].handle;
+				reply.u.mapi_Notify.LogonId = 0;
+				reply.u.mapi_Notify.NotificationType = fnevMbit | fnevSbit | fnevTableModified;
+				reply.u.mapi_Notify.NotificationData.ContentsTableChange.TableEvent = 0x5; /* TABLE_ROW_MODIFED */
+				reply.u.mapi_Notify.NotificationData.ContentsTableChange.ContentsTableChangeUnion.ContentsRowModifiedNotification.FID = folderId;
+				reply.u.mapi_Notify.NotificationData.ContentsTableChange.ContentsTableChangeUnion.ContentsRowModifiedNotification.MID = 0x0;
```

Comment:
These `= 0x0` are not necessary (previous memset) but I guess they are there intentionally for future iterations.